app/services/preserved_object_handler.rb

```diff
@@ -158,8 +158,8 @@ def create_db_objects(status, validated=false)
         pc_attrs[:last_checked_on_storage] = t
       end
       PreservedCopy.create!(pc_attrs)
-      results << result_hash(CREATED_NEW_OBJECT)
     end
+    results << result_hash(CREATED_NEW_OBJECT)
```

Comment:
hmm... though the tests pass, this worries me just a bit, because if the `.create!` call fails with a db error, that'll get trapped by the surrounding `with_active_record_transaction_and_rescue` block, and the result code list would then rightfully have a db failure related result code, in addition to this success code.

that's all based on running this in my head, happy to take a shot at writing a test case for the worry i'm describing if you think i'm off-base.

Reply:
I don't *think* this is the only spot where we add result messages without checking if everything worked;  I would be in favor of it being a separate ticket (patch party?)

Reply:
plus, putting result writing within a transaction doesn't mean the logs don't get written until the transaction is done, does it?

Reply:
I feel like I'm missing something...why move the results message outside the transaction?

Reply:
because I'm foolish??

Reply:
to @ndushay's questions in this thread:

> putting result writing within a transaction doesn't mean the logs don't get written until the transaction is done, does it?

nope, i don't think so.  i believe they'll get written when the calls is made, as usual.

my understanding is that everything inside the transaction happens in real time as it normally would, but the database operations don't get committed (and thus don't become visible to other DB connections) unless the transaction succeeds in its entirety (if it doesn't, then the DB updates are rolled back).

> I don't think this is the only spot where we add result messages without checking if everything worked; I would be in favor of it being a separate ticket (patch party?)

good point... off-hand, i can't think of a spot where this happens, but does seem like a good patch party ticket (at least making sure we don't do that, and cleaning up instances where we do).